apps/OpenSpace/ext/launcher/include/sgctedit/display.h

```diff
@@ -0,0 +1,90 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2022                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_UI_LAUNCHER___DISPLAY___H__
+#define __OPENSPACE_UI_LAUNCHER___DISPLAY___H__
+
+#include <QCheckBox>
+#include <QComboBox>
+#include <QIntValidator>
+#include <QLabel>
+#include <QLayout>
+#include <QLineEdit>
+#include <QPainter>
+#include <QPainterPath>
+#include <QPoint>
+#include <QPushButton>
+#include <QTextBrowser>
+#include <QVector>
+#include <QWidget>
+
+#include <vector>
+#include <iostream>
+
+#include "windowcontrol.h"
+#include "monitorbox.h"
+
+
+class Display : public QWidget
+{
+Q_OBJECT
+public:
+    explicit Display(MonitorBox* monitorRenderBox, std::vector<QRect>& monitorSizeList,
+        const unsigned int nMaxWindows, const QString* winColors);
```

Comment:
The color is passed-in this way because it resides in SgctEdit. Both Display and WindowControl need to get the same color. The const here prevents future editors of this file from changing the color. But if someone wanted to change it at SgctEdit, then it would be changed in both Display and WindowControl, keeping them consistent.

Reply:
In that case it would be easier to understand if it comes in as a `std::array<QColor, 4>` instead that is stored in a `const std::array<QColor, 4>`.  That signals the same (the Display class mustn't change the color) and it fixes the issue of 1) having the color represented as a `QString` and 2) having a pointer to some other person holding the memory, making the understanding of the class non-local